test/component/uk/gov/hmrc/individualincomedesstub/EmploymentSpec.scala

```diff
@@ -191,37 +193,102 @@ class EmploymentSpec extends BaseSpec {
-                     |]}
-     """.stripMargin.replaceAll("\n", "")
+      val request =
+        s"""
+         {
+           "startDate": "2016-01-01",
+           "endDate": "2017-03-01",
+           "payments": [
+               {
+                   "paymentDate": "201601-28",
+                   "taxablePayment": 1000.55,
+                   "nonTaxablePayment": 0
+               }
+           ]
+         }"""
 
       When("I request to create an employment")
       val response = requestCreateEmployment(request)
 
       Then("The response code should be 400 (Bad Request)")
-      response.code shouldBe 400
+      response.code shouldBe BAD_REQUEST
       response.body shouldBe """{"code":"INVALID_REQUEST","message":"paymentDate: invalid date format"}"""
     }
   }
 
+  feature("retrieve employments") {
+
+    scenario("request with an invalid nino") {
+      Given("a request with an invalid nino")
+      val httpRequest = Http(s"$serviceUrl/individuals/ABCDEFGHI/employments/income")
+
+      When("the employments endpoint is invoked")
+      val httpResponse = httpRequest.asString
+
+      Then("the response should be 400 (Bad Request)")
+      httpResponse.code shouldBe BAD_REQUEST
+      httpResponse.body shouldBe """{"code":"INVALID_REQUEST","message":"Malformed nino submitted"}"""
+    }
+
+    scenario("request without a from date") {
+      Given("a request without a from date")
+      val httpRequest = Http(s"$serviceUrl/individuals/$validNino/employments/income?missingFromDate=whatever")
+
+      When("the employments endpoint is invoked")
+      val httpResponse = httpRequest.asString
+
+      Then("the response should be 400 (Bad Request)")
+      httpResponse.code shouldBe BAD_REQUEST
+      httpResponse.body shouldBe """{"code":"INVALID_REQUEST","message":"fromDate is required"}"""
+    }
+
+    scenario("request with a malformed from date") {
+      Given("a request without a from date")
+      val httpRequest = Http(s"$serviceUrl/individuals/$validNino/employments/income?fromDate=01-01-2017")
+
+      When("the employments endpoint is invoked")
+      val httpResponse = httpRequest.asString
+
+      Then("the response should be 400 (Bad Request)")
+      httpResponse.code shouldBe BAD_REQUEST
+      httpResponse.body shouldBe """{"code":"INVALID_REQUEST","message":"fromDate: invalid date format"}"""
+    }
+
+    scenario("request with an invalid date range") {
+      Given("a request with an invalid date range")
+      val httpRequest = Http(s"$serviceUrl/individuals/$validNino/employments/income?fromDate=2017-01-02&toDate=2017-01-01")
+
+      When("the employments endpoint is invoked")
+      val httpResponse = httpRequest.asString
+
+      Then("the response should be 400 (Bad Request)")
+      httpResponse.code shouldBe BAD_REQUEST
+      httpResponse.body shouldBe """{"code":"INVALID_REQUEST","message":"Invalid time period requested"}"""
+    }
+
+    scenario("request with corresponding employments") {
```

Comment:
Apart from the name, I can't see the difference between this test and the previous one